[Docs] auto-import type declarations might not generated in the integration with vue projects

### Package

v4.x

### Description

We know that Nuxt UI uses `unplugin-auto-import` and `unplugin-vue-components` vite plugins, which should generate the files:

- `auto-imports.d.ts`
- `components.d.ts`

on development/production environments.

The problem is that these declaration files are generated by Vite plugins, while the default Vue project boilerplate currently uses the following `build` setup:

```json
{
  "name": "vue-project",
  "version": "0.0.0",
  "private": true,
  "type": "module",
  "scripts": {
    "dev": "vite",
    "build": "run-p type-check \"build-only {@}\" --",
    "preview": "vite preview",
    "build-only": "vite build",
    "type-check": "vue-tsc --build"
  },
  "dependencies": {
    "vue": "^3.5.40"
  }
}
```
The `build` script above runs two processes in **parallel** : type-checking, vite build process. 

This can create a race condition, because `vue-tsc` can possibly start before the Vite plugins, that have generated the required declaration files.
 
For example, a clean build may fail with a message like this:
```
error TS2304: Cannot find name 'useToast'.
```
This is expected because vue-tsc has no knowledge of the Nuxt UI auto-imports until auto-imports.d.ts has been generated.


I think it worth mentioning in Vue specific docs, as this is the default boilerplate of `package.json` that many people [use](https://vuejs.org/guide/quick-start.html).

### Additional context

_No response_

--- a/docs/content/docs/1.getting-started/2.installation/2.vue.md
+++ b/docs/content/docs/1.getting-started/2.installation/2.vue.md
@@ -126,6 +126,19 @@ components.d.ts
 
 ::
 
+::warning
+These declaration files are only written when Vite runs. The default `create-vue` `build` script runs `vue-tsc` in parallel with `vite build` (`run-p type-check "build-only {@}"`), so on a clean checkout type-checking can start before the files exist and fail with errors like `Cannot find name 'useToast'`. Run the type-check after the build instead so the declarations are generated first.
+
+```json [package.json]
+{
+  "scripts": {
+    "build": "run-s build-only type-check"
+  }
+}
+```
+
+::
+
 ::tip
 Internally, Nuxt UI relies on custom alias to resolve the theme types. If you're using TypeScript, you should add an alias to your `tsconfig` to enable auto-completion in your `vite.config.ts`.
 

